tests/test_submit_tuxbuild.py

```diff
@@ -51,102 +61,117 @@ def submit_tuxbuild(self, tuxbuild):
+            "https://raw.githubusercontent.com/Linaro/meta-lkft/sumo/recipes-kernel/linux/files/distro-overrides.config",
+            "https://raw.githubusercontent.com/Linaro/meta-lkft/sumo/recipes-kernel/linux/files/systemd.config",
+            "https://raw.githubusercontent.com/Linaro/meta-lkft/sumo/recipes-kernel/linux/files/virtio.config",
         ]
 
         # Make sure metadata values match expected values
         expected_metadata = {
-            'git_repo': "https://gitlab.com/Linaro/lkft/mirrors/next/linux-next",
-            'git_ref': "master",
-            'git_sha': "5302568121ba345f5c22528aefd72d775f25221e",
-            'git_short_log': '5302568121ba ("Add linux-next specific files for 20201021")',
-            'git_describe': "next-20201021",
-            'kconfig': base_kconfig + ["CONFIG_IGB=y", "CONFIG_UNWINDER_FRAME_POINTER=y"],
-            'kernel_version': "5.9.0",
-            'config': "https://builds.tuxbuild.com/t8NSUfTBZiSPbBVaXLH7kw/config",
-            'download_url': "https://builds.tuxbuild.com/t8NSUfTBZiSPbBVaXLH7kw/",
-            'duration': 541,
-            'toolchain': 'gcc-9',
+            "config": "https://builds.tuxbuild.com/22j6EntJ5Zvge15BqZdxWSJllti/config",
+            "download_url": "https://builds.tuxbuild.com/22j6EntJ5Zvge15BqZdxWSJllti/",
+            "duration": 273,
+            "git_describe": "next-20211224",
+            "git_ref": "master",
+            "git_repo": "https://gitlab.com/Linaro/lkft/mirrors/next/linux-next",
+            "git_sha": "ea586a076e8aa606c59b66d86660590f18354b11",
+            "git_short_log": "ea586a076e8a (\"Add linux-next specific files for 20211224\")",
+            "kconfig": base_kconfig + ["CONFIG_IGB=y", "CONFIG_UNWINDER_FRAME_POINTER=y", "CONFIG_SYN_COOKIES=y"],
+            "kernel_version": "5.16.0-rc6",
+            "toolchain": "gcc-11",
         }
-        for expected_key in expected_metadata.keys():
-            self.assertEqual(expected_metadata[expected_key], getattr(build.metadata, expected_key))
 
-        # Make sure there's no extra attributes in the metadata object
-        metadata_attrs = build.metadata.__dict__
-        del metadata_attrs["id"]
-        self.assertEqual(sorted(expected_metadata.keys()), sorted(metadata_attrs.keys()))
+        for k, v in expected_metadata.items():
+            self.assertEqual(getattr(testrun.metadata, k), v, msg=k)
 
-        environment = self.squad.group("my_group").project("my_project").environment("x86")
+        environment = project.environment("x86_64")
         self.assertIsNotNone(environment)
 
-        suite = self.squad.group("my_group").project("my_project").suite("build")
+        suite = project.suite("build")
         self.assertIsNotNone(suite)
 
-        test = first(self.squad.tests(name="gcc-9-defconfig-b9979cfa"))
-        self.assertEqual("build/gcc-9-defconfig-b9979cfa", test.name)
+        test = first(self.squad.tests(name="gcc-11-defconfig-ec3ad359"))
```

Comment:
Where is this coming from?  is this a fake name that squad-client came up with? since the 'build_name' wasn't filled in build.json?

or wait, this is how it looks now before the new way of doing it with the 'build_name' right?

Reply:
Yes, this is what it looks like now before changing the code.